packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1309,63 +1307,63 @@ List<PluginInterfaceResolution> resolvePlatformImplementation(
 
       // If it hasn't been skipped, it's a candidate for auto-registration, so
       // add it as a possible resolution.
-      final String resolutionKey = getResolutionKey(platform: platformKey, packageName: implementsPackage);
-      possibleResolutions.putIfAbsent(resolutionKey, () => <PluginInterfaceResolution>[]);
-      possibleResolutions[resolutionKey]!.add(PluginInterfaceResolution(
-        plugin: plugin,
-        platform: platformKey,
-      ));
+      possibleResolutions.putIfAbsent(implementsPackage, () => <Plugin>[]);
+      possibleResolutions[implementsPackage]!.add(plugin);
     }
-  }
 
-  // Now resolve all the possible resolutions to a single option for each
-  // plugin, or throw if that's not possible.
-  bool hasResolutionError = false;
-  final List<PluginInterfaceResolution> finalResolution = <PluginInterfaceResolution>[];
-  for (final MapEntry<String, List<PluginInterfaceResolution>> entry in possibleResolutions.entries) {
-    final List<PluginInterfaceResolution> candidates = entry.value;
-    // If there's only one candidate, use it.
-    if (candidates.length == 1) {
-      finalResolution.add(candidates.first);
-      continue;
-    }
-    // Next, try direct dependencies of the resolving application.
-    final Iterable<PluginInterfaceResolution> directDependencies = candidates.where((PluginInterfaceResolution r) {
-      return r.plugin.isDirectDependency;
-    });
-    if (directDependencies.isNotEmpty) {
-      if (directDependencies.length > 1) {
+    final List<Plugin> pluginResolution = <Plugin>[];
```

Comment:
What's the purpose of this intermediate list? It seems like every call to `add` to this list could instead just be a `finalResolution.add` instead.

Reply:
That's right. It's a preliminary step to be able to extract all functionality within the loop to its own method as the logic then is used for resolving native plugins, too:
https://github.com/flutter/flutter/pull/137040/files#diff-68361f6fc9aafa5fa7de042eec4603dd4f321eb783a00edd57780ef78c1094bbR1290

Reply:
It still seems like it could just be an array of `PluginResolutions` in the extracted version too, but the change does make more sense in the extracted version. Thanks.